app/jobs/email_job.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+class EmailJob < ApplicationJob
+  # When the job runs, it will check emails that should have been sent within
+  # EMAIL_CHECK_INTERVAL.
+  # This also implies that the job should be configured to run at the same rate so items
+  # don't get missed and duplicate emails don't get sent.
+  EMAIL_CHECK_INTERVAL = 10.minutes
+
+  # Start with opening reminders.
+  def perform
+    generate_opening_email_items.each do |course, course_emails_hash|
+      course_emails_hash.each do |user_email, items_hash|
+        ConsolidatedMailer.opening_reminder_email(course, user_email, items_hash).deliver_later
+      end
+    end
+  end
+
+  private
+
+  # Generates a nested hash with the following structure for opening reminders:
+  # {
+  #   Course_1 => {
+  #     User::Email_1 => {
+  #       ActableType_1 => [ActableItems...],
+  #       ActableType_2 => [ActableItems...]
+  #     },
+  #     User::Email_2 => {
+  #       ActableType_1 => [ActableItems...],
+  #       ActableType_2 => [ActableItems...]
+  #     }
+  #   },
+  #   Course_2 => {...}
+  # }
+  #
+  # @return [Hash]
+  def generate_opening_email_items
+    ActsAsTenant.without_tenant do
+      opening_lesson_plan_items.each do |item|
+        next unless item.actable.email_enabled?('opening')
+
+        item.course.users.each do |user|
+          emails_hash[item.course][user.emails.first][item.actable_type].push(item.actable)
+        end
+      end
+    end
+    emails_hash
+  end
+
+  def opening_lesson_plan_items
+    datetime_range = (Time.zone.now - EMAIL_CHECK_INTERVAL)..Time.zone.now
+    Course::LessonPlan::Item.where(published: true, start_at: datetime_range).
+      or(Course::LessonPlan::Item.where(created_at: datetime_range))
```

Comment:
Any reasons that we want to check `created_at`?

Reply:
To cover situations where an instructor creates an assessment with a start_at in the past. In that case, the emails should still be sent out to notify the students that something has been created.